README.md

```diff
@@ -245,9 +250,10 @@ interface XRCubeLayer extends XRLayer {
 // Graphics Bindings
 //
 
-dictionary XRWebGLLayerSourceInit {
+dictionary XRLayerInit {
   required unsigned int pixelWidth;
```

Comment:
I am not sure pixelWidth/Height should be marked as 'required' (unless I misunderstand what this actually means). For example, for projection layer it is not required. But scalefactor might be.

Reply:
fixed.
I think scaleFactor should likely move to `getViewSubImage`

Reply:
That would be too late. The scaleFactor only applies to projection layers, and is used when allocating the swapchain, so it has to be supplied at layer creation time. It means "Scale the recommended projection layer size by this much".

Reply:
Oh, and the pixelWidth/Height IS required for non-projection layers, which in the current IDL is the only place that dictionary is used. Projection layers were using the old XRWebGLLayerInit dict.

Reply:
I was under the impression that this particular scalefactor would tell you by how much to upscale your rendered area.
For instance, you might want to allocate your cylinder layer quite large in case it is close to the user but when the user walks away, you want to optimize the experience and maybe render at half the resolution.

In that case you'd pass scalefactor of .5 and get back a smaller viewport. Then you render to a smaller area and when the swap happens, the compositor will know by how much to upscale.

This would only be useful for cylinder/quad layers I suspect.

Reply:
> Oh, and the pixelWidth/Height IS required for non-projection layers, which in the current IDL is the only place that dictionary is used. Projection layers were using the old XRWebGLLayerInit dict.

Yes! I will change it back